src/Layout/Footer.tsx

```diff
@@ -239,21 +235,14 @@ function LanguageModal({
             <RadioGroup
               aria-label="language"
               name="language"
-              value={language}
+              value={i18n.language}
               onChange={(event) => {
-                setlanguage(event.target.value);
-                localStorage.setItem("language", event.target.value);
-                if (
-                  projectDetails &&
-                  projectDetails.purpose !== "planet-cash-signup" &&
-                  router.query.to
-                ) {
-                  loadPaymentSetup({
-                    projectGUID: router.query.to as string,
-                    paymentSetupCountry: country,
-                    shouldSetPaymentDetails: false,
-                  });
-                }
+                const lang = event.target.value;
+                const { pathname, query, asPath } = router;
+                // set cookie to store user preferences
+                document.cookie = `NEXT_LOCALE=${lang}; max-age=31536000; path=/`;
```

Comment:
I am wondering if the language can not be switched otherwise as directly manipulating the cookie, e.g. with `i18n.changeLanguage(lang);` or some functionality provided by `next-i18next`?

Reply:
The cookie is only to persist the user preference (if they explicitly select a language). i18n.changeLanguage would not accomplish that. Also, the cookie is accessible by middleware, whereas localStorage isn't.

The priority for language shown is
1. Language specified by the URL locale param
2. Language specified by the user via NEXT_LOCALE
3. Autodetecting the locale based on the accept-language header sent by the browser.

This solution seems best supported by Next js.